bots/pr/src/main/java/org/openjdk/skara/bots/pr/LimitedCensusInstance.java

```diff
@@ -55,27 +64,43 @@ private static Namespace namespace(Census census, String hostNamespace) {
         if (namespace == null) {
             throw new RuntimeException("Namespace not found in census: " + hostNamespace);
         }
-
         return namespace;
     }
 
-    private static Optional<JCheckConfiguration> configuration(HostedRepositoryPool hostedRepositoryPool, HostedRepository remoteRepo, String name, String ref) throws IOException {
-        return hostedRepositoryPool.lines(remoteRepo, Path.of(name), ref).map(JCheckConfiguration::parse);
+    private static Optional<JCheckConfiguration> configuration(HostedRepository remoteRepo, String name, String ref) {
+        Optional<List<String>> conf;
+        try {
+            conf = Optional.of(Arrays.stream(remoteRepo.fileContents(name, ref).split("\n")).toList());
+        } catch (RuntimeException e) {
+            if (e.getMessage().contains("Request returned bad status:")) {
+                if (!e.getMessage().contains("404")) {
+                    throw e;
+                }
+            }
+            conf = Optional.empty();
+        }
+        return conf.map(JCheckConfiguration::parse);
     }
 
-    private static Optional<JCheckConfiguration> jCheckConfiguration(HostedRepositoryPool hostedRepositoryPool,
+    private static Optional<JCheckConfiguration> jCheckConfiguration(
             HostedRepository repository, String ref, HostedRepository confOverrideRepo, String confOverrideName,
-            String confOverrideRef) throws IOException {
+            String confOverrideRef) throws IOException, InvalidJCheckConfException {
         Optional<JCheckConfiguration> configuration;
-        if (confOverrideRepo == null) {
-            configuration = configuration(hostedRepositoryPool, repository, ".jcheck/conf", ref);
-        } else {
-            configuration = configuration(hostedRepositoryPool,
-                    confOverrideRepo,
-                    confOverrideName,
-                    confOverrideRef);
+        try {
+            if (confOverrideRepo == null) {
+                configuration = configuration(repository, ".jcheck/conf", ref);
+            } else {
+                configuration = configuration(confOverrideRepo,
+                        confOverrideName,
+                        confOverrideRef);
+            }
+            return configuration;
+        } catch (RuntimeException e) {
+            if (e.getMessage().contains("Request returned bad status:")) {
+                throw e;
+            }
+            throw new InvalidJCheckConfException(e);
```

Comment:
Here, if we meet some `uncheckedException`, we just throw it, otherwise it will be treated as `InvalidJCheckException` as well.